core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -900,19 +875,17 @@ DescendingIter descendingIter(K from, boolean inclusive) {
             int handle = getEntryField(next, OFFSET_HANDLE_INDEX);
 
             int compare = -1;
-            if (next != Chunk.NONE) {
-                compare = compare(from, readKey(next));
-            }
+            if (next != Chunk.NONE)
+                compare = (-1) * comparator.compareSerializedKeyAndKey(readKey(next), from);
```

Comment:
The reason is that the comparator for serialized key and (non-serialized) key has reverse order on the arguments than the order used here (you can notice the flipped calling order).
Note that the original comparator already has the sign flip multiplication in it, so this is the only way to negate it. 
As a side note, I think the comparator trick is unsafe and we should see if it can be refactored outside the scope of this PR.

Reply:
Can we please have a comment here (about the reason for -1)?

Reply:
switched argument order and added a comment inside the mixedComparator to clarify things